src/current/v23.2/select-for-update.md

```diff
@@ -16,22 +16,50 @@ The following diagram shows the supported syntax for the optional `FOR` locking
+| `FOR UPDATE` | Acquire an exclusive lock on the rows returned by the [`SELECT`]({% link {{ page.version.version }}/selection-queries.md %}) statement. For details, see [`FOR UPDATE` usage](#for-update-usage).  
 
-Locking strength dictates the row-level locking behavior on rows retrieved by a `SELECT` statement.
+Under `SERIALIZABLE` isolation:
 
-Parameter | Description
-----------|------------
-`FOR SHARE`/`FOR KEY SHARE` | This syntax is a no-op, allowed for PostgreSQL compatibility. Specifying `FOR SHARE`/`FOR KEY SHARE` does not cause CockroachDB to use shared locks over the rows retrieved by a statement.<br><br>Note that CockroachDB always [ensures serializability]({% link {{ page.version.version }}/demo-serializable.md %}), regardless of the specified locking strength.
-`FOR UPDATE`/`FOR NO KEY UPDATE` | Lock the rows returned by the [`SELECT`]({% link {{ page.version.version }}/selection-queries.md %}) statement, such that other transactions trying to access the rows must wait for the transaction to finish.<br><br>Note that in CockroachDB, the `FOR NO KEY UPDATE` locking strength is identical to the `FOR UPDATE` locking strength.
+- Shared locks are not enabled by default. To enable shared locks for `SERIALIZABLE` transactions, configure the [`enable_shared_locking_for_serializable` session setting]({% link {{ page.version.version }}/session-variables.md %}). To perform [foreign key]({% link {{ page.version.version }}/foreign-key.md %}) checks under `SERIALIZABLE` isolation with shared locks, configure the [`enable_implicit_fk_locking_for_serializable` session setting]({% link {{ page.version.version }}/session-variables.md %}). This matches the default [`READ COMMITTED`]({% link {{ page.version.version }}/read-committed.md %}) behavior.
+
+- If the [`optimizer_use_lock_op_for_serializable` session setting]({% link {{ page.version.version }}/session-variables.md %}) is enabled, the [cost-based optimizer]({% link {{ page.version.version }}/cost-based-optimizer.md %}) uses a `Lock` operator to construct query plans for `SELECT` statements using the `FOR UPDATE` and `FOR SHARE` clauses. This more closely matches the PostgreSQL behavior, but will create more round trips from [gateway node]({% link {{ page.version.version }}/architecture/sql-layer.md %}#gateway-node) to [replica leaseholder]({% link {{ page.version.version }}/architecture/replication-layer.md %}#leases) in some cases.
+
+### Lock strengths
+
+Lock "strength" determines how restrictive the lock is to concurrent transactions attempting to access the same row.
+
+{% include {{ page.version.version }}/sql/select-lock-strengths.md %}
+
+Note that CockroachDB [ensures serializability]({% link {{ page.version.version }}/demo-serializable.md %}) regardless of the specified lock strength.                    
```

Comment:
nit: maybe this should be qualified as "Note that CockroachDB ensures serializability _when using SERIALIZABLE isolation_ regardless of the specified lock strength."